### PR TITLE
[WIP] Style Guide

### PR DESCRIPTION
**Pull request by iamanai**
Creating a style guide file that is external to the actual wiki. The reasoning behind this is that since all wiki editing is done through GitHub now, having a style guide page on the wiki itself is redundant.

Currently the style guide is an almost 1:1 copy of the one presented in #6. I can make additional changes before merging if necessary.

For previous discussions related to this PR, see #104.

### CODE REVIEW

**Comment by RaikireHiuduo** [comment]
Just a note that https://github.com/ppy/osu-wiki/tree/master/wiki/Article_Style_Guide exists inside the `wiki` folder. Two conflicting guide is not a good sign. Please go read and salvage whatever left in the old Article Style Guide that still can be used in the new one.

If you plan to put it inside the `wiki` folder, you can clear all the `<a>` HTML tags and TOC. Replace the `<br>` tags with two newline spaces to let Markdown do its job instead.

**Comment by iamanai** [comment]
Personally I wanted it outside the wiki folder because we'll never be viewing it inside the wiki itself (well, you could, but why?). But I do agree that this would be more consistent and cleaner.

I'll put something in the readme as well.

**Comment by sr229** [comment]
@RaikireHiuduo  the endgame here is to truncate the old wiki styling guide on the Article Style Guide folder and replace it with a final version of #6 's draft.

I think you should do that @iamanai

**Comment by iamanai** [comment]
That's along the lines of what I was going to do, except I wasn't going to put it in the wiki.

Is pretty much everyone in agreement of nuking the old style guide and replacing it with a touched-up version of #6?

**Comment by RaikireHiuduo** [comment]
I am actually thinking less of nuking but more of _assimilation_ of old and new.

Yeah, sure.

File: README.md [hunk 1]
```diff
@@ -14,3 +14,7 @@ The majority of content in this repository is licensed under [CC-BY-NC 4.0](http
 As this is a wiki, there may be content with third party licences. These licences will be cited local to the content, and override the global licence file.
 
 Please note that this *does not cover* the usage of the "osu!" or "ppy" branding in any software, resources, advertising or promotion, as this is protected by trademark law. If you require clearance for the use of these terms, please [contact us](mailto:contact@ppy.sh).
+
+# Contributing
+
+Contributions to the osu!wiki are done through GitHub. A basic guide to using GitHub, specifically tailored for the osu!wiki project, is available [here](/wiki/osu-contribution-guide/en.md). The article style guide, which _must_ be adhered to for all new / rewritten pages going forward, is available [here](/wiki/Article-Style-Guide/en.md).
```
Comment (MegaApplePi) [inline README.md]: Do not use `en.md` when linking to articles, the back end handles this

File: wiki/Article_Style_Guide/en.md [hunk 2]
```diff
@@ -1,192 +1,278 @@
...
+Bold&Italic : **_BI_**
+Quote       : > rhythm is just a "click" away!
+Redirect    : [page text](link "Alt text"), or  [myreference]: http://www.google.com "Omnipotent searchbar"
+                                                ...
+                                                [page text][my-reference]
+Image       : Add an exclamation mark, '!', in front.
+              ![broken text](link "Alt text"), or   [broken text]: http://www.google.com "Omnipotent searchbar"
+                                                    ...
+                                                    ![broken text][my-reference]
+              Alternatively, clickable image (from @MegaApplePi).
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: there is no need to include my username here

File: wiki/Article_Style_Guide/en.md [hunk 3]
```diff
@@ -1,192 +1,278 @@
...
+
+## Introduction
+**Restructure or rewrite the whole page to GitHub-flavoured Markdown(GFM)-compliance.** HTML tags are out-of-question.
+
+GFM can be learned in one day, but do not expect too much out of it as it intended for plaintext with bare-minimum formatting needed to function as a readable file for a webpage.
+
+### Language support
+**_osu!web [as of 14/1/2017. Still WIP]_**  
+Link: <https://new.ppy.sh/wiki/>
+1. HTML tags not supported (any HTML tags and its contents will be removed from public view).
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: add a line before this line

File: wiki/Article_Style_Guide/en.md [hunk 4]
```diff
@@ -1,192 +1,278 @@
...
+2. Image display is literal display without proper encapsulation.
+
+**_Not supported (in GitHub view):_**
+1. HTML5 inline CSS (specifically, "style"/"class" attribute) standard,
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: ditto (add a line before this line)

File: wiki/Article_Style_Guide/en.md [hunk 5]
```diff
@@ -1,192 +1,278 @@
...
+  - Capitalisation and punctuation is strict (must be spelt exactly or _404 not found_) when directing to the page.
+    - Linking to osu!catch: `/wiki/Game_Modes/osu!catch/`
+  - For directing to a specific header in the page, the header name must be in lowercase and spaces/punctuations are to be replaced by hyphen (`-`).
+    - Linking to specific section of FAQ: `/wiki/FAQ#ive-never-played-this-ouendan-thing---what-is-it/`
+- `https://` preferred.
+- When translating, keep the **english link reference name**.
+  - This is to ease maintenance in the event of image/link problem (direct copy and paste, rather than hunting for that one link to be fixed).
+
+**_Keyboard keys:_**
+- Use grave marks (\`) and [camelCase](https://en.wikipedia.org/wiki/Camel_case) format.
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: use:
```
- Use grave marks (`` ` ``) and [camelCase](https://en.wikipedia.org/wiki/Camel_case) format.
```

File: wiki/Article_Style_Guide/en.md [hunk 6]
```diff
@@ -1,192 +1,278 @@
...
+  - Example 1: `;`
+  - Example 2: `D`
+  - Example 3: `Spacebar`
+  - Example 4: `leftCtrl`
+  - Example 5: `Shift`-`1`
+
+**_Page study of interest:_**
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: idk, is this really needed?

File: wiki/Article_Style_Guide/en.md [hunk 7]
```diff
@@ -1,192 +1,278 @@
...
+- [FAQ](https://github.com/ppy/osu-wiki/tree/master/wiki/FAQ/en.md) {general structure, image structure style (same level)}
+- [Multiplayer](https://github.com/ppy/osu-wiki/tree/master/wiki/Multiplayer/en.md) {general structure, raw HTML table usage, image structure style (same level)}
+- [Skinning](https://github.com/ppy/osu-wiki/tree/master/wiki/Skinning/en.md) {Image structure style (img folder)}
+- [Accuracy](https://github.com/ppy/osu-wiki/tree/master/wiki/Accuracy/en.md) {Image structure style (same level)}
+- [osu!store](https://github.com/ppy/osu-wiki/tree/master/wiki/osu!store/en.md) {Extreme Markdown and HTML hybrid}
+
+## General styling
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: Might as well make each item a subheading of themselves to include more examples (and make it look a bit more cleaner)

File: wiki/Article_Style_Guide/en.md [hunk 8]
```diff
@@ -1,192 +1,278 @@
...
+Unordered   : Use '-' styling.
+Ordered     : Use '1. /2. /3. ' styling.
+Code        : ```Multiline```, `Single`
+Escaping    : Use backslash, \
+```
+
+## Link
+### General
+Assuming the fact that the file structure right now is _not a permanent file structure_ (subject to restructuring).
+- Absolute linking from root to source if possible (`/wiki/{toSourcePoint}/`). Exercise caution when using relative linking (`./{sameLevelSourcePoint}`).
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: in the form of technicalities, this is a relative link with an absolute path

File: wiki/Article_Style_Guide/en.md [hunk 9]
```diff
@@ -1,192 +1,278 @@
...
+- Reference style preferred over inline style (reference style is reusable).
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: is it? I refer inline unless the link is used multiple times.

File: wiki/Article_Style_Guide/en.md [hunk 10]
```diff
@@ -1,192 +1,278 @@
...
+
+[thumb1]: link-to-thumb.png "Click to view full size"
+[full1]: link-to-full.png
+...
+[![alt-text][thumb1]][full1]
+```
+
+**_Image storage:_**
+- Put **images relevant only to the page** at the **same level as the page itself**.
+  - For older images, put in a deprecated folder (?).
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: I don't think this is a standard (yet?). Most deprecated images just sit on my PC until I see a fit to use them in HOO.

File: wiki/Article_Style_Guide/en.md [hunk 11]
```diff
@@ -1,192 +1,278 @@
...
+  - .jpg for thumbnail image unless it has transparency.
+  - .gif for animation and icons.
+- For new image update, replace the old image using the same name and extension as the old image.
+  - Put old images to deprecated folder(?).
+- No idea about all the images from the MediaWiki by time of writing.
+  - May need to manually salvage the images from the old osu!wiki yourself.
+
+## Table styling
+Based from #70 :  
+Use Markdown table. Most of the Markdown will work except the unordered/ordered listing (must use HTML tags, see: <https://stackoverflow.com/questions/19950648/how-to-write-lists-inside-a-markdown-table>).
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: Just to say, if you have to put a list inside a table... you are overdoing both the table and the list

File: wiki/Article_Style_Guide/en.md [hunk 12]
```diff
@@ -1,192 +1,278 @@
...
+(Please note that the above three lines have two trailing whitespaces intentionally)
+
+P.S. A neat feature if you are using Github's Atom; you can type `table` then press ENTER and it will generate a template GFM table for you.
+```Markdown
+| Header One     | Header Two     |
+| :------------- | :------------- |
+| Item One       | Item Two       |
+```
+
+## Unconfirmed
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: move this section to the bottom and call it `Miscellaneous`

File: wiki/Article_Style_Guide/en.md [hunk 13]
```diff
@@ -1,192 +1,278 @@
...
+- git simple cheatsheet: https://rogerdudler.github.io/git-guide/
+- Basic of forking: https://kbroman.org/github_tutorial/pages/branching.html
+- "-s ours" (**Merge OWN branch ONLY**, ignore other branch commits; solves back commit issues _when done correctly_): https://stackoverflow.com/a/729723
+
+Note: Yes, go get [git](https://git-scm.com/ "git homepage") into your current OS and learn git terminal (GUI is possible but good luck). Doing it online is possible for simple immediate updates but if you want to go all the way, using git is the way to go.
+
+P.S. Study how to fork, pull(fetch/merge), push, add, commit, branch (local and remote), pull request, and basic vim terminal commands (git's default text processor). Optionally, learn how to squash commits _only if you have time_, **properly**.
+
+---
+## Et Cetera
```
Comment (MegaApplePi) [inline wiki/Article_Style_Guide/en.md]: make this the `Miscellaneous` section

**Comment by Repflez** [comment]
#129 and #155 have been made improvements on the guide. You're still welcome to suggest even more changes but for now, closing this PR.

**Pull request closed without merging**